lib/ansible/modules/packaging/os/package_facts.py

```diff
@@ -62,9 +67,26 @@
   type: complex
   contains:
     packages:
-      description: list of dicts with package information
+      description:
+        - Maps the package name to a non-empty list of dicts with package information.
+        - Every dict in the list corresponds to one installed version of the package.
+        - The fields described below are present for all package managers. Depending on the
+          package manager, there might be more fields for a package.
       returned: when operating system level package manager is specified or auto detected manager
       type: dict
+      contains:
+        name:
+          description: The package's name.
+          returned: always
+          type: str
+        version:
+          description: The package's version.
+          returned: always
+          type: str
+        source:
+          description: Where information on the package came from.
+          returned: always
+          type: str
       sample_rpm:
```

Comment:
Would not an example be displayed on the module's documentation if `sample` were used instead of `sample_rpm` ? I guess `sample_deb` could be kept as is (and won't be displayed at all).

Another way: keep `sample_rpm` and `sample_deb` untouched and add a `sample` entries containing only the documented keys (`name`, `source`, `version`).

Reply:
Good point! I added a stripped down version of `sample_rpm`, which contains only the documented keys. (Disadvantage: both kernels have the same version, they only differ in keys not shown.)